### DISCUSSION TITLE
How to reset the login password

### DISCUSSION THREAD

**Discussion by chaptergy** [NginxProxyManager/nginx-proxy-manager#1634] [FAQs / Instructions]
If you have forgotten your password and need to reset it here is how you do it. This depends on what database you are using. If you have two docker containers one of which is a database, you are using MySQL, if you just have the single npm docker container, you probably use the SQLite database. Since November of 2021, the example docker-compose file is using SQLite, prior to that is was MySQL.

### Step 1
Run these commands on the machine running the docker container containing the database but replacing `<container-name>` with the name of the docker container containing your npm instance (It will most likely be something like `nginxproxymanager_core`):
```
docker exec -it <container-name> sh
apt update && apt install sqlite3 -y
sqlite3 /data/database.sqlite
```
You have now entered the SQL mode, where you set the status of all users to deleted:
```sql
UPDATE user SET is_deleted=1;
.exit
exit
```
### Step 2
If your NPM container has been running, restart it. If it has not been running, start it now.

### Step 3
You have now created an admin user you can login with by accessing your NPM in the browser and logging in with the default login information:
**login:** admin@example.com
**pass:** `changeme`

### Step 4
You can now either just use this user, or you can re-enable the old account and use the new account to change the password of the old one. To re-enable it, once again execute the following commands:
```commandline
docker exec -it <container-name> sh
sqlite3 /data/database.sqlite
```

Replace `<container-name>` again, as in step 1. Then set all users to not deleted by running:

```sql
UPDATE user SET is_deleted=0;
.exit
exit
```

### Step 1
Run these commands on the machine running the docker container containing the database but replacing `<container-name>` with the name of the docker container containing your database (It will most likely be something like `nginxproxymanager_db_1`):
```
docker exec -it <container-name> sh
mysql -u root -p
```  
You will have to enter the root password of your database. You have set it in your docker-compose file through the [`MYSQL_ROOT_PASSWORD` variable](https://github.com/jc21/nginx-proxy-manager/blob/master/docs/setup/README.md?plain=1#L84). Then continue entering the following commands:

```sql
USE npm;
UPDATE user SET is_deleted=1;
quit
exit
```
### Step 2
If your NPM container has been running, restart it. If it has not been running, start it now.

### Step 3
You have now created an admin user you can login with by accessing your NPM in the browser and logging in with the default login information:
**login:** admin@example.com
**pass:** `changeme`

### Step 4
You can now either just use this user, or you can re-enable the old account and use the new account to change the password of the old one. To re-enable all previously disabled users, once again execute the following commands:
```commandline
docker exec -it <container-name> sh
mysql -u root -p
```

Replace `<container-name>` and enter the root password again, as in step 1.

```sql
USE npm;
UPDATE user SET is_deleted=0;
quit
exit
```

<sup>[Based on comment by talesam](https://github.com/jc21/nginx-proxy-manager/issues/230#issuecomment-815078355)</sup>

**Comment by telometto** [comment]
A lifesaver!

**Comment by stealthdave** [comment]
I was able to use this to get access to Nginx Proxy Manager, but when I logged in, none of my existing proxies showed up.  After a little investigation, I discovered that this is because we are creating a _second_ user and all of the proxies were owned by the _first_ user.  Luckily, there is an easy fix which is updating the `own_user_id` to the new user, which in my case was `2`, for the host and stream tables.  Assuming your new user id is `2`, run the following SQL statements.

```SQL
UPDATE dead_host SET owner_user_id=2;
UPDATE proxy_host SET owner_user_id=2;
UPDATE redirection_host SET owner_user_id=2;
UPDATE stream SET owner_user_id=2;
```

This will update the tables to be owned by the new user.

**Reply by the-polak** [reply]
Thanks for adding this comment.  Should I be concerned about a weird message when accessing SSL CERTIFICATES:

Cannot read properties of null (reading 'avatar')

**Reply by littlewhitefox** [reply]
you can try 
```sql
UPDATE certificate SET owner_user_id=2;
````

**Reply by tiagojulianoferreira** [reply]
Just collaborating with the thread, I retrieved the ACLs for the new account using:

```
UPDATE access_list SET owner_user_id=2;
```

**Reply by sirjmann92** [reply]
Lifesaver! Freaked out when I logged in and saw all of my stuff missing with the avatar error. I had deleted the original user so I could re-use the credentials so I was worried it would be unrecoverable. Glad I was wrong. Thanks for the help!

**Reply by alekshnl** [reply]
Alternatively you can use the admin@example.com account to reset the password in the users section. After completion don't forget to delete the admin@example.com account again.

**Comment by TK-3375** [comment]
When I reach step 3 it says that there is no "relevant user found" when I try to login (either with my useraccount, or the default one). Ideas ?

**Reply by RubenSibon** [reply]
Have you really restarted the docker container? I.e.: `docker-compose restart`?

**Reply by TK-3375** [reply]
I use portainer but I guess it's the same. Yes I have stopped and restarted the container (the entire server too, to be safe). No luck

**Reply by Blackwidow-sudo** [reply]
yes i can confirm that: Step 4 & 5 didn't work for me either. I get the same response.

**Reply by carlitoselmago** [reply]
in my case this happened cose I already had other users, I had to manually delete the extra users before and then it worked

**Comment by TK-3375** [comment]
> yes i can confirm that: Step 4 & 5 didn't work for me either. I get the same response.

For whatever reason i tried again and it worked... I have no idea what changed since i first tried. I guess a few restarts helped somehow

**Reply by Blackwidow-sudo** [reply]
oh ok. Well i just deleted the data folder. I had no proxies set up anyway.

**Comment by mzagozen** [comment]
An alternative that does not require creating a new user and losing any data is to just update the password hash for the user. The passwords are hashed with *bcrypt*. You can use Python to create a new hash:

```
$ python -c 'import bcrypt; print(bcrypt.hashpw("password".encode("utf-8"), bcrypt.gensalt()).decode("utf-8"))'
$2b$12$tMOjHhjExkb8d/EMrHOOy.WA0WAznst6WZyt7a1828FUOvEb9NhtW
```

Then you can install `sqlite3` in the container like in Step 1 and update the hash for any user:

```
UPDATE auth SET secret = '<hash>' WHERE user_id = <admin-user-id>
```

**Reply by thunermay** [reply]
Nice idea! How do you get the correct user id?

UPDATE:
`SELECT id, email FROM user;` should to the trick. Just look up the user by the email and set the hash accordingly (column `nickname` optionally is also available)

**Comment by 1lIiiIl1** [comment]


**Reply by TobiasGdot** [reply]
your problem seems to be, you haven't followed the whole instruction nor read through the other comments ;)
Use the new account to reset the password of the old one or you gonna have to update the owner ids of the tables holding the content

**Comment by IMNotMax** [comment]
For those who, like me have this password issue in Nginx Proxy Manager ADD-ON in Home Assistant, here some information.
1/ First, have a SSH access to your HA server (Advanced SSH & Web Terminal is great with Protection Mode DISABLED !)

2/ Ssh your HA and type : `docker ps` to find the NPM container name
```
Welcome to Home Assistant OS.
Use ha to access the Home Assistant CLI.

**docker ps**

CONTAINER ID   IMAGE                                                              COMMAND                  CREATED          STATUS                PORTS                                                                                                                          NAMES
110c2ac207ed   ghcr.io/hassio-addons/nginxproxymanager/aarch64:1.0.1              "/init"                  5 minutes ago    Up 5 minutes          0.0.0.0:80-81->80-81/tcp, :::80-81->80-81/tcp, 0.0.0.0:443->443/tcp, :::443->443/tcp                                           addon_a0d7b954_nginxproxymanager
fa23a1250422   homeassistant/aarch64-addon-mariadb:2.6.1                          "/init"                  22 minutes ago   Up 22 minutes                                                                                                                                        addon_core_mariadb
567c019a872f   zigbee2mqtt/zigbee2mqtt-aarch64:1.35.2-1                           "/sbin/tini -- /dock…"   3 hours ago      Up 3 hours            0.0.0.0:8485->8485/tcp, :::8485->8485/tcp                      
```

**_mine is `addon_a0d7b954_nginxproxymanager `_**

3/ Enter the container using `docker exec -it addon_a0d7b954_nginxproxymanager sh`
Now your are in the container, and NOW you can think of editing the DB... but mysql or sqlite are not installed...

4/ just type `apk add sqlite`
And now follow ALL the steps (copied from above and adapted)

5/ `sqlite3 /config/database.sqlite` to open the DB

6/ find your user id (usually it's 1 because you only have 1 user
`SELECT id, email, is_deleted FROM user WHERE email='your-user-email-here'`

7/ Change the is_deleted from 0 to 1 (this will force the admin@example - changeme user to be activated
`UPDATE user SET is_deleted = 1 WHERE id = 1` (caution, put the id of your user, but usually it's 1)

8/ NOW, in home assistant, restart the add-on Nginx Proxy Manager and login with `admin@example.com` and `changeme`

9/ create a new admin user (DON'T FORGET THIS PASSWORD TOO !!!)

10/ logout

11/ Come back to your container via SSH and open the DB again (steps 2 to 5)

12/ Change is_deleted from 1 to 0 `UPDATE user SET is_deleted = **0** WHERE id = 1` (if your user id is 1 of course)

13/ Restart the add-on in Home Assitant

14/ Log in with the new admin user you just created
**NOW AND ONLY NOW** you can change the password of your previous user (look in the Users tab).

15/ logout, and login with your old user and newly created password, and delete the admin user created in this procedure.

Hope this helps


NB: The update NGINX Proxy Manager Add-on (v1) does not rely on MariaDB anymore, that's why you have to edit the SQLite database inside the NginxProxyManager Container

**Reply by de-sascha** [reply]
Wohoo! Thanks it saved my evening today. Thanks God.

**Comment by flippet** [comment]
The sqlite solution deleted all my proxy redirections and in the SSL section. Also saw an error "Owner:null" which wouldn't go away. 

So at the end I had no other choice than deleting the entire deployment and creating it again from scratch.

**Comment by jalvarezz13** [comment]
This should be in the documentation! Thanks :)

**Comment by DooMFeaR** [comment]
> If you have forgotten your password and need to reset it here is how you do it. This depends on what database you are using. If you have two docker containers one of which is a database, you are using MySQL, if you just have the single npm docker container, you probably use the SQLite database. Since November of 2021, the example docker-compose file is using SQLite, prior to that is was MySQL.
> 
> ## SQLite (default since November 2021)
> ### Step 1
> Run these commands on the machine running the docker container containing the database but replacing `<container-name>` with the name of the docker container containing your npm instance (It will most likely be something like `nginxproxymanager_core`):
> 
> ```
> docker exec -it <container-name> sh
> apt update && apt install sqlite3 -y
> sqlite3 /data/database.sqlite
> ```
> 
> You have now entered the SQL mode, where you set the status of all users to deleted:
> 
> ```sql
> UPDATE user SET is_deleted=1;
> .exit
> exit
> ```
> 
> ### Step 2
> If your NPM container has been running, restart it. If it has not been running, start it now.
> 
> ### Step 3
> You have now created an admin user you can login with by accessing your NPM in the browser and logging in with the default login information: **login:** [admin@example.com](mailto:admin@example.com) **pass:** `changeme`
> 
> ### Step 4
> You can now either just use this user, or you can re-enable the old account and use the new account to change the password of the old one. To re-enable it, once again execute the following commands:
> 
> ```
> docker exec -it <container-name> sh
> sqlite3 /data/database.sqlite
> ```
> 
> Replace `<container-name>` again, as in step 1. Then set all users to not deleted by running:
> 
> ```sql
> UPDATE user SET is_deleted=0;
> .exit
> exit
> ```
> 
> ## MySQL (default prior to November 2021)
> ### Step 1
> Run these commands on the machine running the docker container containing the database but replacing `<container-name>` with the name of the docker container containing your database (It will most likely be something like `nginxproxymanager_db_1`):
> 
> ```
> docker exec -it <container-name> sh
> mysql -u root -p
> ```
> 
> You will have to enter the root password of your database. You have set it in your docker-compose file through the [`MYSQL_ROOT_PASSWORD` variable](https://github.com/jc21/nginx-proxy-manager/blob/master/docs/setup/README.md?plain=1#L84). Then continue entering the following commands:
> 
> ```sql
> USE npm;
> UPDATE user SET is_deleted=1;
> quit
> exit
> ```
> 
> ### Step 2
> If your NPM container has been running, restart it. If it has not been running, start it now.
> 
> ### Step 3
> You have now created an admin user you can login with by accessing your NPM in the browser and logging in with the default login information: **login:** [admin@example.com](mailto:admin@example.com) **pass:** `changeme`
> 
> ### Step 4
> You can now either just use this user, or you can re-enable the old account and use the new account to change the password of the old one. To re-enable all previously disabled users, once again execute the following commands:
> 
> ```
> docker exec -it <container-name> sh
> mysql -u root -p
> ```
> 
> Replace `<container-name>` and enter the root password again, as in step 1.
> 
> ```sql
> USE npm;
> UPDATE user SET is_deleted=0;
> quit
> exit
> ```
> 
> [Based on comment by talesam](https://github.com/jc21/nginx-proxy-manager/issues/230#issuecomment-815078355)

Works great to login with the default credentials, but after I change the email and password and I reboot the container, I get the error "Invalid password".  Even if I add a new user and password, reboot the container, everything doesn't work anymore for some reason.

**Comment by lahirusamishka** [comment]
# ✅ Fix: Nginx Proxy Manager Login Issue & Restore Proxy Hosts

If you’re locked out and need to reset your admin password and recover missing proxy hosts, follow these simple steps:

---

### 🔄 Steps to Reset Login and Show Hosts Again

1. **Open the container shell:**
   ```bash
   docker exec -it <container-name> sh
   ```

2. **Install sqlite3 (if needed):**
   ```bash
   apt update && apt install sqlite3 -y
   ```

3. **Open the database:**
   ```bash
   sqlite3 /data/database.sqlite
   ```

4. **Mark all users as deleted:**
   ```sql
   UPDATE user SET is_deleted=1;
   .exit
   exit
   ```

5. **Restart the container.**

6. **Log in with default admin:**
   - **Email:** `admin@example.com`
   - **Password:** `changeme`

7. **Proxy hosts will be missing — fix it:**

   - Enter the container again:
     ```bash
     docker exec -it <container-name> sh
     sqlite3 /data/database.sqlite
     ```

   - Run this to restore users:
     ```sql
     UPDATE user SET is_deleted=0;
     .exit
     exit
     ```

8. **Log in again**, reset the original user's password, and all proxy hosts will reappear.

---

📝 *Tested and worked for me — hope this helps someone!*

**Comment by Aladdin4u** [comment]
I was able to fix it without deleting all the users in SQLite. I forgot my email.
nginx proxy manager creates two folders by default inside your project (letsencrypt and data).
Inside the data folder is the database.sqlite, which can be viewed using the cat command `cat database.sqlite` to list all tables and rows

**Comment by kiloptero** [comment]
I can access the database , where is the password? Jajjaaj

**Reply by IMNotMax** [reply]
You should not have to type a password with direct access with sqlite3

**Reply by IMNotMax** [reply]
![output](https://github.com/user-attachments/assets/49e34a09-1fd7-434e-befc-56aa3e2e4f9b)

**Reply by Aladdin4u** [reply]
Password is under auth table.

The Password is encrypted inside the secret column.

You can delete the data inside auth table, NPM will ask you for a new password

**Reply by IMNotMax** [reply]
Holly Molly I did not understand well his issue lol

**Comment by kiloptero** [comment]
reinstall and restote backup jeje

Andres Roepke

**Comment by GAS85** [comment]
Amazing 🚀 

This is short how-to disable two-factor (2fa) authentication if somehow it does not work any more as e.g. in #5451 

1. Do db backup.
2. Open db as per **Step 1**:
```bash
docker exec -it <container-name> sh
apt update && apt install sqlite3 -y
sqlite3 /data/database.sqlite
```
3. Verify exact row:
```sql
SELECT id, user_id, type, meta
FROM auth
WHERE user_id = 1 AND type = 'password' AND is_deleted = 0;
```
You should see something like:
```plain
1|1|password|{"totp_secret":"YOUR_SECRET_","totp_enabled":1,"totp_enabled_at":"2026-01-27T08:39:09.474Z","backup_codes":["REDACTED"]}
```
4.  Disable TOTP. SQLite's JSON functions can modify just totp_enabled:
```sql
UPDATE auth
SET meta = json_set(meta, '$.totp_enabled', false),
    modified_on = datetime('now')
WHERE user_id = 1
  AND type = 'password'
  AND is_deleted = 0;
```
5. Verify
```sql
SELECT id, user_id, json_extract(meta, '$.totp_enabled') AS totp_enabled
FROM auth
WHERE user_id = 1 AND type = 'password' AND is_deleted = 0;
```
If output is:
```plain
1|1|0
```
You good. Exit db with `.quit` command.
6. Restart Nginx Proxy Manager:
```bash
docker restart <container-name>
```
